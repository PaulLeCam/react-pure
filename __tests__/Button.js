import React from 'react';
import TestRenderer from 'react-test-renderer';

import Button from '../src/Button';

describe('Button', () => {
  it('renders a button by default', () => {
    const renderer = TestRenderer.create(<Button />);
    expect(renderer.toJSON).toMatchSnapshot();
  });

  it('renders an anchor the `href` property is set', () => {
    const renderer = TestRenderer.create(<Button href="#" />);
    expect(renderer.toJSON).toMatchSnapshot();
  });

  it('transfers the `href` property to the anchor', () => {
    const renderer = TestRenderer.create(<Button href="#test" />);
    expect(renderer.toJSON).toMatchSnapshot();
  });

  it('adds the `pure-button` class', () => {
    const renderer = TestRenderer.create(<Button className="my-button" />);
    expect(renderer.toJSON).toMatchSnapshot();
  });

  it('adds the `pure-button-active` class if the `active` property if set', () => {
    const renderer = TestRenderer.create(<Button active />);
    expect(renderer.toJSON).toMatchSnapshot();
  });

  it('transfers the `disabled` and adds the `pure-button-disabled` class if the `disabled` property if set', () => {
    const renderer = TestRenderer.create(<Button disabled />);
    expect(renderer.toJSON).toMatchSnapshot();
  });

  it('adds the `pure-button-primary` class if the `primary` property if set', () => {
    const renderer = TestRenderer.create(<Button primary />);
    expect(renderer.toJSON).toMatchSnapshot();
  });
});
