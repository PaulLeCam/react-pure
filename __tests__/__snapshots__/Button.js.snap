// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button adds the \`pure-button\` class 1`] = `[Function]`;

exports[`Button adds the \`pure-button-active\` class if the \`active\` property if set 1`] = `[Function]`;

exports[`Button adds the \`pure-button-primary\` class if the \`primary\` property if set 1`] = `[Function]`;

exports[`Button renders a button by default 1`] = `[Function]`;

exports[`Button renders an anchor the \`href\` property is set 1`] = `[Function]`;

exports[`Button transfers the \`disabled\` and adds the \`pure-button-disabled\` class if the \`disabled\` property if set 1`] = `[Function]`;

exports[`Button transfers the \`href\` property to the anchor 1`] = `[Function]`;
